src/data/labels.js

```diff
@@ -0,0 +1,326 @@
+            throw new Error('boardId is not valid.');
+        }
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        if(board === null){
+            throw new Error('There is not board with that id.');
+        }
+
+        const cardList = board.cards;
+        let foundCard;
+        let returnLabel;
+        
+        if(cardList.length === 0){
+            throw new Error('That board has no cards.');
+        }
+
+        for(let card of cardList){
+            if(card._id.toString() === cardId){
+                foundCard = card;
+            }
+        }
+        if(!foundCard){
+            throw new Error('Card not found');
+        } 
+
+        let index = 0;
+        for(let label of foundCard){
+            if(label._id.toString() === labelId){
+                returnLabel = label;
+                break;
+            }
+            index++;
+        }
+        if(!returnLabel){
+            throw new Error('Label not found');
+        } 
+
+        return index;
+
+    },
+
+    /**
+     * Get a specific card from a board.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @returns A card object
+     */
+    async getCardById(boardId, cardId){
+        if(!boardId || !error_handler.checkObjectId(boardId)){
+            throw new Error('boardId is not valid');
+        }
+        if(!cardId || !error_handler.checkObjectId(cardId)){
+            throw new Error('cardId is not valid');
+        }
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        if(board === null){
+            throw new Error('There is not board with that id.');
+        }
+        let foundCard;
+        
+        for(let card of board.cards){
+            if(card._id.toString() === cardId){
+                foundCard = card;
+            }
+        }
+        if(!foundCard){
+            throw new Error('Card not found.');
+        } 
+        foundCard._id = foundCard._id.toString();
+
+        return foundCard;
+    }, 
+    /**
+     * Add a label to a card.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @param {string} text The label's text.
+     * @param {string} color The label's color hex code.
+     * @returns A label object.
+     */
+    async addLabel(boardId, cardId, text, color){
+        if(!boardId || !error_handler.checkObjectId(boardId)){
+            throw new Error('boardId is not valid.');
+        }
+        if(!cardId || !error_handler.checkObjectId(cardId)){
+            throw new Error('cardId is not valid.');
+        }
+        if(!text || !error_handler.checkNonEmptyString(text)){
+            throw new Error('Text must not be empty.');
+        }
+        if(!color || !error_handler.checkColor(color)){
```

Comment:
color may have to be randomly selected from pre-approved values when we actually implement

Reply:
I can add this feature if we all want it, but I was thinking they would select the color when making the label.